common/http/http.go

```diff
@@ -39,7 +39,14 @@ func PostJSON(
 
 	res, err := httpClient.Do(req.WithContext(ctx))
 	if res != nil {
-		defer (func() { err = res.Body.Close() })()
+		defer (func() {
+			finalErr := res.Body.Close()
```

Comment:
I'm not sure that the body here needs to be closed at all. Apart from the fact that in case of a mistake "res" does not have to be considered at all.
SEE: https://github.com/golang/go/blob/master/src/net/http/request.go#L175

Reply:
Isn't it required, to prevent resource leak?

> From the http.Client documentation:
> 
>  If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close. If the Body is not both read to EOF and closed, the Client's underlying RoundTripper (typically Transport) may not be able to re-use a persistent TCP connection to the server for a subsequent "keep-alive" request.

Reply:
In that case we can check for err being nil and then have the defer statement. No need to check for res!=nil